packages/lib/src/core/core.ts

```diff
@@ -165,7 +165,7 @@ class Core {
          * When PaymentMethod is defined as a string - retrieve a component from the componentsMap and recall this function passing in a valid class
          */
         if (typeof PaymentMethod === 'string' && paymentMethods[PaymentMethod]) {
-            return this.handleCreate(paymentMethods[PaymentMethod], options);
+            return this.handleCreate(paymentMethods[PaymentMethod], { type: PaymentMethod, ...options });
```

Comment:
I don't get how this `type` is then used?
It's just passed into the constructor for the PaymentMethod and then...?

Reply:
This is basically injecting the string used to create the component as a prop. This will then be available on the component under `this.props.type`. `UIElement` uses this on the `.type` getter:

```js
    /**
     * Return the type of an element
     */
    get type(): string {
        return this.props.type || this.constructor['type'];
    }
```